eng/scripts/job-matrix/README.md

```diff
@@ -165,11 +165,124 @@ once for each instance of a language version.
+        "TestFromSource": [ false, true ]
+    }
+}
+
+### example-matrix.json to import
+{
+    "matrix": {
+      "operatingSystem": [ "windows", "linux" ],
+      "client": [ "netty", "okhttp" ]
+    },
+    "include": [
+        {
+          "operatingSystem": "mac",
+          "client": "netty"
+        }
+    ]
+}
+```
+
+1. The base matrix is generated (sparse in this example):
+    ```
+    {
+      "storage_18": {
+        "endpointType": "storage",
+        "JavaVersion": "1.8"
+      },
+      "cosmos_111": {
+        "endpointType": "cosmos",
+        "JavaVersion": "1.11"
+      }
+    }
+    ```
+1. The imported base matrix is generated (sparse in this example):
+    ```
+    {
+      "windows_netty": {
+        "operatingSystem": "windows",
+        "client": "netty"
+      },
+      "linux_okhttp": {
+        "operatingSystem": "linux",
+        "client": "okhttp"
+      }
+    }
+    ```
 1. Includes/excludes from the imported matrix get applied to the imported matrix
-1. The base matrix is multipled by the imported matrix
-1. Includes/excludes from the top-level matrix get applied to the multiplied matrix
+    ```
+    {
+      "windows_netty": {
+        "operatingSystem": "windows",
+        "client": "netty"
+      },
+      "linux_okhttp": {
+        "operatingSystem": "linux",
+        "client": "okhttp"
+      },
+      "mac_netty": {
+        "operatingSystem": "mac",
+        "client": "netty"
+      }
+    }
+    ```
+1. The base matrix is multipled by the imported matrix (in this case, the base matrix has 2 elements, and the imported
+   matrix has 3 elements, so the product is a matrix with 6 elements:
+    ```
+      "storage_18_windows_netty": {
+        "endpointType": "storage",
+        "JavaVersion": "1.8",
+        "operatingSystem": "windows",
+        "client": "netty"
+      },
+      "storage_18_linux_okhttp": {
+        "endpointType": "storage",
+        "JavaVersion": "1.8",
+        "operatingSystem": "linux",
+        "client": "okhttp"
+      },
+      "storage_18_mac_netty": {
+        "endpointType": "storage",
+        "JavaVersion": "1.8",
+        "operatingSystem": "mac",
+        "client": "netty"
+      },
+      "cosmos_111_windows_netty": {
+        "endpointType": "cosmos",
+        "JavaVersion": "1.11",
+        "operatingSystem": "windows",
+        "client": "netty"
+      },
+      "cosmos_111_linux_okhttp": {
+        "endpointType": "cosmos",
+        "JavaVersion": "1.11",
+        "operatingSystem": "linux",
+        "client": "okhttp"
+      },
+      "cosmos_111_mac_netty": {
+        "endpointType": "cosmos",
+        "JavaVersion": "1.11",
+        "operatingSystem": "mac",
+        "client": "netty"
+      }
+    }
+    ```
+1. Includes/excludes from the top-level matrix get applied to the multiplied matrix, so the below element will be added
+   to the above matrix, for an output matrix with 7 elements:
+    ```
+    "windows_18_TestFromSource": {
```

Comment:
I also don't see this include in your sample matrix.

Reply:
Ah whoops, forgot to fully copy the json I was testing against back into the markdown...

Reply:
fixed both readme issues